src/Microsoft.Azure.WebJobs.Host/Blobs/Listeners/ScanBlobScanLogHybridPollingStrategy.cs

```diff
@@ -0,0 +1,243 @@
+        }
+
+        public void Notify(IStorageBlob blobWritten)
+        {
+            ThrowIfDisposed();
+            _blobsFoundFromScanOrNotification.Enqueue(blobWritten);
+            _pollLogStrategy.Notify(blobWritten);
+        }
+
+        public void Dispose()
+        {
+            if (!_disposed)
+            {
+                _pollLogStrategy.Dispose();
+                _cancellationTokenSource.Dispose();
+                _disposed = true;
+            }
+        }
+
+        private void ThrowIfDisposed()
+        {
+            if (_disposed)
+            {
+                throw new ObjectDisposedException(null);
+            }
+        }
+
+        public async Task<IEnumerable<IStorageBlob>> PollNewBlobsAsync(
+            IStorageBlobContainer container, ContainerScanInfo containerScanInfo, CancellationToken cancellationToken)
+        {
+            IEnumerable<IStorageListBlobItem> currentBlobs;
+            IStorageBlobResultSegment blobSegment;
+            int blobPollLimitPerContainer = _scanBlobLimitPerPoll / _scanInfo.Count;
+            BlobContinuationToken continuationToken = containerScanInfo.ContinuationToken;
+
+            // if starting the cycle, keep the current time stamp to be used as curser
+            if (continuationToken == null)
+            {
+                containerScanInfo.CurrentSweepCycleStartTime = DateTime.UtcNow;
+            }
+            try
+            {
+                blobSegment = await container.ListBlobsSegmentedAsync(prefix: null, useFlatBlobListing: true, 
+                    blobListingDetails: BlobListingDetails.None, maxResults: blobPollLimitPerContainer, currentToken: continuationToken, 
+                    options: null, operationContext: null, cancellationToken: cancellationToken);
+                currentBlobs = blobSegment.Results;
+            }
+            catch (StorageException exception)
+            {
+                if (exception.IsNotFound())
+                {
+                    return Enumerable.Empty<IStorageBlob>();
+                }
+                else
+                {
+                    throw;
+                }
+            }
+
+            List<IStorageBlob> newBlobs = new List<IStorageBlob>();
+
+            // Type cast to IStorageBlob is safe due to useFlatBlobListing: true above.
+            foreach (IStorageBlob currentBlob in currentBlobs)
+            {
+                cancellationToken.ThrowIfCancellationRequested();
+
+                IStorageBlobProperties properties = currentBlob.Properties;
+                DateTime lastModifiedTimestamp = properties.LastModified.Value.UtcDateTime;
+
+                if (lastModifiedTimestamp > containerScanInfo.LastSweepCycleStartTime)
+                {
+                    newBlobs.Add(currentBlob);
+                }
+            }
+
+            // record continuation token for next chunk retrieval
+            containerScanInfo.ContinuationToken = blobSegment.ContinuationToken;
```

Comment:
When running in a classic SDK AlwaysOn scenario, jobs are expected to be very long lived, so this continuation token is key. In cases where we're running in a free site that may get put to sleep after a short period of inactivity (functions), for large containers we'll likely **never get to the end** - we'll keep restarting at the beginning and chugging through to the point where we get idled out. Meaning blobs added at the end of the container might never get discovered. Note that this is the case even before your changes.